[BUG] step.sendSignal return type leaks internal InngestApi namespace and causes error TS2742 for consumers

**Describe the bug**

Starting in `inngest@4.2.5`, any consumer project that compiles with `declaration: true` (or `composite: true`) and uses `inngest.createFunction(...)` fails to emit `.d.ts` with Typescript error TS2742:

```
error TS2742: The inferred type of '<name>' cannot be named without a reference to
'../../node_modules/inngest/api/api'. This is likely not portable.
A type annotation is necessary.
```

The leak comes from `step.sendSignal()`'s return type, which was changed from `Promise<null>` to `Promise<InngestApi.SendSignalResponse>` in #1479. `InngestApi` is defined in `packages/inngest/src/api/api.ts`, which is **not** in the package's public `exports` map, so TypeScript can't write a portable module specifier for it when emitting consumer declaration files. It falls back to a relative `node_modules/inngest/api/api` reference, which trips TS2742.

This is a clear regression since the same consumer code emits fine on `4.2.4` and earlier.

**To Reproduce**

This issue can be demonstrated in one file (plus a `tsconfig.json`):

`src/index.ts`:
```ts
import { Inngest } from "inngest";

const inngest = new Inngest({ id: "demo" });

export const fn = inngest.createFunction(
  { id: "my-fn", triggers: [{ event: "foo" }] },
  async (ctx) => ctx,
);
```

`tsconfig.json`:
```jsonc
{
  "compilerOptions": {
    "declaration": true,
    "outDir": "dist",
    "module": "Preserve",
    "moduleResolution": "Bundler",
    "strict": true,
    "target": "ES2022"
  },
  "include": ["src"]
}
```

```bash
$ npm i inngest@4.2.6
$ npx tsc

src/index.ts:5:14 - error TS2742: The inferred type of 'fn' cannot be named
without a reference to '../../node_modules/inngest/api/api'. This is likely
not portable. A type annotation is necessary.
```

The function does not even need to call `step.sendSignal` since `step` is part of the inferred handler type returned by `createFunction`.  So the leak cascades into every consumer's `.d.ts` regardless of whether they use the signal API.

**Expected behavior**

`tsc` emits the consumer's `.d.ts` cleanly, as it did in `4.2.4` and earlier (when `step.sendSignal` returned `Promise<null>`). The TypeScript compiler should be able to resolve any type referenced from the SDK's publicly exported API.

**TL;DR**

Just need a one-line addition to `packages/inngest/src/index.ts`:

```diff
 export * from "@inngest/ai";
 export type { StandardSchemaV1 } from "@standard-schema/spec";
+export type { InngestApi } from "./api/api.ts";
 export { experiment } from "./components/ExperimentStrategies.ts";
```

**Verified leak path**

In the published `4.2.5+` `.d.ts` tree, `InngestApi` reaches the public surface at exactly two points:

- `node_modules/inngest/components/InngestStepTools.d.ts:216`
  ```ts
  sendSignal: (idOrOptions: StepOptionsOrId, opts: SendSignalOpts) => Promise<InngestApi.SendSignalResponse>;
  ```
- `node_modules/inngest/components/Inngest.d.ts:178`
  ```ts
  }): Promise<InngestApi.SendSignalResponse>;
  ```

Both files import `InngestApi` from `../api/api.js`, but `./api/api` (or `./api`) is not in `package.json`'s `exports` map, so consumer-side declaration emit can't write a portable specifier for `InngestApi.SendSignalResponse`.

**Code snippets / Logs / Screenshots**

```
$ git log --all --oneline -S "Promise<InngestApi.SendSignalResponse>" --reverse
6b9769ca EXE-266: Update sendSignal's return type (#1479)

$ git tag --contains 6b9769ca
inngest@4.2.5
inngest@4.2.6
```

Regression commit: **#1479** ("EXE-266: Update sendSignal's return type"), first released in **`inngest@4.2.5`**. The PR's checklist notes "Added unit/integration tests: N/A additional testing not needed for typing change" — the consumer declaration-emit path wasn't covered, which is why it slipped through.

(For context, the existing `pnpm test:composite` does call `inngest.sendSignal(...)` but discards the result with `void`, so TypeScript never has to *emit* `InngestApi.SendSignalResponse` to a portable `.d.ts`. The leak would have been caught if the result were exported, or if the test exercised `step.sendSignal` from inside a `createFunction` handler.)

**Some Workarounds I Tried**

The simplest approach was a single annotation:

```ts
const fn: InngestFunction.Any = inngest.createFunction(/* ... */);
```

However, it turns out this does _not_ work because the annotation back-propagates as the contextual type into the `createFunction` call and collapses the handler generic to `Handler.Any`. Inside the handler, `step.run("id", () => ({ a: 1 }))` then returns `unknown` instead of `{ a: 1 }`, downstream `.filter(i => ...)` / `.map(i => ...)` lose their parameter types, and consumers see a cascade of secondary TS7006 errors. Als note that `as InngestFunction.Any` casting had the same back-propagation behavior (and is forbidden by my strict lint rules anyway).

The only workaround I could find that preserves handler inference was a two-variable indirection at every call site (which was not fun to deal with in a few dozen places across our codebase):

```ts
const _x = inngest.createFunction(opts, handler); // full handler inference
const x: InngestFunction.Any = _x;                // portable outer type
export { x };
```

This is verbose, has to be applied at every `createFunction` call site, and is the kind of thing that probably shouldn't be required of SDK consumers in the first place.

**Simplest fix (as mentioned above)**

A one-line addition to `packages/inngest/src/index.ts`:

```diff
 export * from "@inngest/ai";
 export type { StandardSchemaV1 } from "@standard-schema/spec";
+export type { InngestApi } from "./api/api.ts";
 export { experiment } from "./components/ExperimentStrategies.ts";
```
I made this change locally and installed the package directly from my local inngest-js fork and I can confirm all TypeScript inference errors across our monorepo were resolved. 

Once `InngestApi` is reachable through the public package entry, TypeScript writes `import("inngest").InngestApi.SendSignalResponse` into consumer `.d.ts` files instead of the non-portable relative reference and then the TS2742 error goes away.

Also note that `export type` (rather than `export`) was intentional.  That ensures only the namespace member types (`SendSignalResponse`, etc.) need to be reachable for inference.  The `InngestApi` class itself is internal and isn't used by consumers. `export type` erases at runtime, so there's no bundle-size impact either.

This is the same pattern already used for similar cases. PR #1432 ("Fix composite project type exports", `inngest@4.1.1`) added these missing type re-exports for the same reasons:

- `StepFetch`, `GetFunctionOutputRaw`, `GroupTools`, `GroupToolsDeps`, `Realtime`, `HandlerWithTriggers`, `DurationLike`, `InstantLike`, `ZonedDateTimeLike`, `ExclusiveKeys`, `JsonifyObject`

`InngestApi` was simply missed when #1479 introduced a new public reference to it.

**System info**

- npm package version: `inngest@4.2.5` and `inngest@4.2.6` (current)
- Last working version: `inngest@4.2.4`
- TypeScript: 5.9.3
- Node: 24.15.0
- Framework: framework-agnostic (reproduces with the bare SDK)
- Platform: any (this is a build-time TypeScript declaration emit issue, unrelated to runtime)

--- a/.changeset/busy-drinks-drop.md
+++ b/.changeset/busy-drinks-drop.md
@@ -1,0 +1,5 @@
+---
+"inngest": patch
+---
+
+Fix `TS2742` errors in projects that use `declaration` or `composite` `true` by exporting `SendSignalResponse` appropriately.

--- a/packages/inngest/src/api/api.ts
+++ b/packages/inngest/src/api/api.ts
@@ -12,6 +12,7 @@ import {
   type OutgoingOp,
   ok,
   type Result,
+  type SendSignalResponse,
 } from "../types.ts";
 import {
   type BatchResponse,
@@ -78,15 +79,6 @@ export namespace InngestApi {
   export interface SendSignalOptions {
     signal: string;
     data?: unknown;
-  }
-
-  export interface SendSignalResponse {
-    /**
-     * The ID of the run that was signaled.
-     *
-     * If this is undefined, the signal could not be matched to a run.
-     */
-    runId: string | undefined;
   }
 }
 
@@ -267,7 +259,7 @@ export class InngestApi {
     options?: {
       headers?: Record<string, string>;
     },
-  ): Promise<Result<InngestApi.SendSignalResponse, ErrorResponse>> {
+  ): Promise<Result<SendSignalResponse, ErrorResponse>> {
     const url = await this.getTargetUrl("/v1/signals");
 
     const body = {
@@ -292,7 +284,7 @@ export class InngestApi {
       .then(async (res) => {
         // A 404 is valid if the signal was not found.
         if (res.status === 404) {
-          return ok<InngestApi.SendSignalResponse>({
+          return ok<SendSignalResponse>({
             runId: undefined,
           });
         }

--- a/packages/inngest/src/components/Inngest.ts
+++ b/packages/inngest/src/components/Inngest.ts
@@ -53,6 +53,7 @@ import {
   type MetadataTarget,
   type SendEventOutput,
   type SendEventResponse,
+  type SendSignalResponse,
   sendEventResponseSchema,
 } from "../types.ts";
 import { getAsyncCtx } from "./execution/als.ts";
@@ -585,7 +586,7 @@ export class Inngest<const TClientOpts extends ClientOptions = ClientOptions>
      * multiple systems together using branch names.
      */
     env?: string;
-  }): Promise<InngestApi.SendSignalResponse> {
+  }): Promise<SendSignalResponse> {
     const headers: Record<string, string> = {
       ...(env ? { [headerKeys.Environment]: env } : {}),
     };
@@ -601,7 +602,7 @@ export class Inngest<const TClientOpts extends ClientOptions = ClientOptions>
     signal: string;
     data?: unknown;
     headers?: Record<string, string>;
-  }): Promise<InngestApi.SendSignalResponse> {
+  }): Promise<SendSignalResponse> {
     const res = await this.inngestApi.sendSignal(
       { signal, data },
       { ...this.headers, ...headers },

--- a/packages/inngest/src/components/InngestStepTools.ts
+++ b/packages/inngest/src/components/InngestStepTools.ts
@@ -2,7 +2,6 @@ import { type AiAdapter, models } from "@inngest/ai";
 import type { StandardSchemaV1 } from "@standard-schema/spec";
 import { z } from "zod/v3";
 
-import type { InngestApi } from "../api/api.ts";
 import type { Jsonify } from "../helpers/jsonify.ts";
 import { normalizeEventMeta } from "../helpers/sessions.ts";
 import { timeStr } from "../helpers/strings.ts";
@@ -24,6 +23,7 @@ import {
   type OutgoingOp,
   type ReceivedEventMeta,
   type SendEventOutput,
+  type SendSignalResponse,
   StepMode,
   StepOpCode,
   type StepOptions,
@@ -614,7 +614,7 @@ export const createStepTools = <
       (
         idOrOptions: StepOptionsOrId,
         opts: SendSignalOpts,
-      ) => Promise<InngestApi.SendSignalResponse>
+      ) => Promise<SendSignalResponse>
     >(
       ({ id, name }, opts) => {
         return {

--- a/packages/inngest/src/index.ts
+++ b/packages/inngest/src/index.ts
@@ -119,6 +119,7 @@ export type {
   RegisterOptions,
   ScheduledTimerEventPayload,
   SendEventBaseOutput,
+  SendSignalResponse,
   StepOptions,
   StepOptionsOrId,
   TimeStr,

--- a/packages/inngest/src/types.ts
+++ b/packages/inngest/src/types.ts
@@ -818,6 +818,15 @@ export type SendEventOutput<TOpts extends ClientOptions> = Omit<
 export type SendEventOutputWithMiddleware<_TOpts extends ClientOptions> =
   SendEventBaseOutput;
 
+export interface SendSignalResponse {
+  /**
+   * The ID of the run that was signaled.
+   *
+   * If this is undefined, the signal could not be matched to a run.
+   */
+  runId: string | undefined;
+}
+
 /**
  * Discriminator for which output transform to extract from middleware.
  */

--- a/packages/inngest/test/composite_project/src/index.ts
+++ b/packages/inngest/test/composite_project/src/index.ts
@@ -149,6 +149,15 @@ export const fnWithStepRun = inngest.createFunction(
   },
 );
 
+// Exercise step.sendSignal to ensure SendSignalResponse is portable in
+// declaration emit.
+export const fnWithSendSignal = inngest.createFunction(
+  { id: "my-fn-send-signal", triggers: [{ event: "foo" }] },
+  async ({ step }) => {
+    return step.sendSignal("signal", { signal: "foo", data: { foo: "bar" } });
+  },
+);
+
 // Exercise eventType and typed event schemas to ensure StandardSchemaV1
 // and related trigger types are portable.
 export const myEventType = Inngest.eventType("app/user.created", {

--- a/packages/inngest/test/composite_project/tsconfig.json
+++ b/packages/inngest/test/composite_project/tsconfig.json
@@ -1,6 +1,5 @@
 {
   "compilerOptions": {
-    "baseUrl": ".",
     "target": "es5",
     "forceConsistentCasingInFileNames": true,
     "composite": true,

